tf_yarn/examples/pytorch_distributed_example.py

```diff
@@ -0,0 +1,127 @@
+import os
+import getpass
+from uuid import uuid4
+
+import torch
+import torchvision
+
+import torch.distributed as dist
+
+import torch.nn as nn
+import torch.nn.functional as F
+
+from tf_yarn.distributed import client
+from tf_yarn.distributed.task import get_task
+from tf_yarn.topologies import TaskSpec, NodeLabel
+from tf_yarn.pytorch import model_ckpt
+
+import cluster_pack
+
+
+def train_fcn():
+    local_rank = 0  # TBD later
+    _, rank, size, master_addr, master_port, _ = get_task()
+    print(f'master: {master_addr}:{master_port}')
+
+    model_path = f"viewfs://root/user/{getpass.getuser()}/tf-yarn-distribute-example_{str(uuid4())}"
+
+    # Initialization of Pytorch distributed bakend
+    os.environ['MASTER_ADDR'] = master_addr
+    os.environ['MASTER_PORT'] = str(master_port)
+    dist.init_process_group(backend='nccl', world_size=size, rank=rank)
+
+    nb_epoch = 50
+
+    # ######### Training
+    # Network def
+    class Net(torch.nn.Module):
+        def __init__(self):
+            super(Net, self).__init__()
+            self.conv1 = nn.Conv2d(1, 32, 3, 1)
+            self.conv2 = nn.Conv2d(32, 64, 3, 1)
+            self.dropout1 = nn.Dropout(0.25)
+            self.dropout2 = nn.Dropout(0.5)
+            self.fc1 = nn.Linear(9216, 128)
+            self.fc2 = nn.Linear(128, 10)
+
+        def forward(self, x):
+            x = self.conv1(x)
+            x = F.relu(x)
+            x = self.conv2(x)
+            x = F.relu(x)
+            x = F.max_pool2d(x, 2)
+            x = self.dropout1(x)
+            x = torch.flatten(x, 1)
+            x = self.fc1(x)
+            x = F.relu(x)
+            x = self.dropout2(x)
+            x = self.fc2(x)
+            output = F.log_softmax(x, dim=1)
+            return output
+
+    # Set of exclusive GPU for the process
```

Comment:
No support to use several GPUs on a GPU machine with several GPUs ?

Reply:
FAR